architecture/smg/runbooks/07-docker-data-drive-not-mounted.md

```diff
@@ -0,0 +1,55 @@
+# Docker data drive not mounted
+
+By default the servers provided by OpenStack for the instance size used by Datalabs are
+only provisioned with a 16Gb drive. This is insufficient to store the large Notebook
+images that Datalabs uses.
+
+To resolve this issue, all servers are provisioned with a second 100Gb drive and Docker
+is configured to use this instead of the default location for images.
+
+In the case where the drive has not been mounted, the directory will be created by Docker
+and it will work without issue until the root drive reaches 85% capacity causing a
+Kubernetes panic and wide spread pod evictions.
+
+This issue can be resolved by the following steps
+
+1. Cordon the node `kubectl cordon <node-name>`.
+1. Drain the node `kubectl drain <node-name> --ignore-daemonsets`.
+1. Login to the node and elevate permissions `sudo su`.
+1. Stop the kubelet and docker services.
+
+    ```bash
+    systemctl stop kubelet.service
+    systemctl stop docker
+    ```
+
+1. Move the `/data` directory to `/data-old`.
+
+    ```bash
+    mv /data /data-old
+    ```
+
+1. Mount the drive using an ansible playbook.
```

Comment:
Is there an existing Ansible playbook that could be referenced here?

Reply:
Ha, I knew you'd ask that. Yes and no, I tweaked an existing one but could probably make a maintenance version